core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -993,6 +993,7 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
       val (outputMetrics, bytesWrittenCallback) = initHadoopOutputMetrics(context)
 
       val writer = format.getRecordWriter(hadoopContext).asInstanceOf[NewRecordWriter[K,V]]
+      require(writer != null, s"Unable to obtain NewRecordWriter")
```

Comment:
There's no string interpolation here, and there is no actual `NewRecordWriter` class (it's an alias). Don't worry I can edit that before merge. I will have Jenkins test just for completeness